react/ui-components/src/atoms/OTPInput.js

```
@@ -45,20 +45,19 @@ const OTPInput = ({
 
   const handlePaste = (e) => {
+    e.preventDefault();
     const pastedData = e.clipboardData.getData("text").slice(0, length);
     const newOtp = [...otp];
     let pasteIndex = 0;
     for (let i = 0; i < newOtp.length && pasteIndex < pastedData.length; i++) {
       const char = pastedData[pasteIndex];
 
       if (
-        (type === "numeric" && !isNaN(char)) ||
+        (type === "numeric" && !isNaN(char) && char !== " ") ||
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)